src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -2814,12 +2819,8 @@ void returnAfter(MethodInformation methodInfo) {
             // special considerations for std::string without adapter
             out.print(");\n" + indent + "rptr = rstr.c_str()");
         }
-        if (adapterInfo != null) {
-            if (methodInfo.allocator || methodInfo.arrayAllocator) {
-                suffix = ", 1, NULL)" + suffix;
-            } else if (!methodInfo.returnType.isPrimitive()) {
-                suffix = ")" + suffix;
-            }
+        if ((methodInfo.allocator || methodInfo.arrayAllocator || !methodInfo.returnType.isPrimitive()) && adapterInfo != null) {
```

Comment:
Since the returnType of allocator is void, we don't need to check explicitly and can revert completely to 
```suggestion
        if (!methodInfo.returnType.isPrimitive() && adapterInfo != null) {
```

Reply:
These is a `!` before `methodInfo.returnType.isPrimitive()`. I believe this check is here because in `returnBefore` we ignore the adapter if the type is a non-void primitive.

Reply:
Ah, void.class.isPrimitive() returns true, interesting...